src/freshrssapi.cpp

```diff
@@ -0,0 +1,546 @@
+			strchr(newflags.c_str(), star_flag[0]) != nullptr) {
+			success = star_article(guid, true);
+		} else if (strchr(oldflags.c_str(), star_flag[0]) != nullptr &&
+			strchr(newflags.c_str(), star_flag[0]) == nullptr) {
+			success = star_article(guid, false);
+		}
+	}
+
+	if (share_flag.length() > 0) {
+		if (strchr(oldflags.c_str(), share_flag[0]) == nullptr &&
+			strchr(newflags.c_str(), share_flag[0]) != nullptr) {
+			success = share_article(guid, true);
+		} else if (strchr(oldflags.c_str(), share_flag[0]) != nullptr &&
+			strchr(newflags.c_str(), share_flag[0]) == nullptr) {
+			success = share_article(guid, false);
+		}
+	}
+
+	return success;
+}
+
+bool FreshRssApi::star_article(const std::string& guid, bool star)
+{
+	std::string token = get_new_token();
+	std::string postcontent;
+
+	if (star) {
+		postcontent = strprintf::fmt(
+				"i=%s&a=user/-/state/com.google/starred&ac=edit&T=%s",
+				guid,
+				token);
+	} else {
+		postcontent = strprintf::fmt(
+				"i=%s&r=user/-/state/com.google/starred&ac=edit&T=%s",
+				guid,
+				token);
+	}
+
+	std::string result = post_content(
+			cfg->get_configvalue("freshrss-url") + FRESHRSS_API_EDIT_TAG_URL,
+			postcontent);
+
+	return result == "OK";
+}
+
+bool FreshRssApi::share_article(const std::string& guid, bool share)
+{
+	std::string token = get_new_token();
+	std::string postcontent;
+
+	if (share) {
+		postcontent = strprintf::fmt(
+				"i=%s&a=user/-/state/com.google/broadcast&ac=edit&T=%s",
+				guid,
+				token);
+	} else {
+		postcontent = strprintf::fmt(
+				"i=%s&r=user/-/state/com.google/broadcast&ac=edit&T=%s",
+				guid,
+				token);
+	}
+
+	std::string result = post_content(
+			cfg->get_configvalue("freshrss-url") + FRESHRSS_API_EDIT_TAG_URL,
+			postcontent);
+
+	return result == "OK";
+}
+
+std::string FreshRssApi::post_content(const std::string& url,
+	const std::string& postdata)
+{
+	std::string result;
+	curl_slist* custom_headers{};
+
+	CURL* handle = curl_easy_init();
+	utils::set_common_curl_options(handle, cfg);
+	add_custom_headers(&custom_headers);
+	curl_easy_setopt(handle, CURLOPT_HTTPHEADER, custom_headers);
+	curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, my_write_data);
+	curl_easy_setopt(handle, CURLOPT_WRITEDATA, &result);
+	curl_easy_setopt(handle, CURLOPT_POSTFIELDS, postdata.c_str());
+	curl_easy_setopt(handle, CURLOPT_URL, url.c_str());
+	curl_easy_perform(handle);
+	curl_easy_cleanup(handle);
+	curl_slist_free_all(custom_headers);
+
+	LOG(Level::DEBUG,
+		"FreshRssApi::post_content: url = %s postdata = %s result = %s",
+		url,
+		postdata,
+		result);
+
+	return result;
+}
+
+rsspp::Feed FreshRssApi::fetch_feed(const std::string& id, CURL* cached_handle)
+{
+	rsspp::Feed feed;
+	feed.rss_version = rsspp::Feed::FRESHRSS_JSON;
+
+	const std::string query = strprintf::fmt("%s?n=%u",
+			id,
+			cfg->get_configvalue_as_int("freshrss-min-items"));
+
+	CURL* handle;
+	if (cached_handle) {
+		// Never seems to be a cached handle
```

Comment:
You mean the function is never passed a non-null pointer? It should be non-null when you're reloading all feeds (e.g. via `auto-reload`, or by pressing `R`): https://github.com/newsboat/newsboat/blob/f7d8ab4ac8d85b6947b150ff8b3388c89dd9f730/src/reloader.cpp#L237-L244.

Reply:
Oh ok, I must have missed that in my testing.

Reply:
I should note that I merely examined the code, not tested it. So if your testing shows that the pointer is `NULL` even when using `R` or `auto-reload`, please file an issue and we'll see where my examination went wrong!